jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt

```diff
@@ -71,12 +72,18 @@ class CodeWhispererRecommendationManager {
         val reformattedReferences = rangeMarkers.map {
             val currentRange = it.key
             val originalReference = it.value
+            var spanEndOffset = currentRange.endOffset
+            var lastChar = runReadAction { currentRange.document.getText(TextRange(spanEndOffset - 1, spanEndOffset)) }
+            while (lastChar == "\n") {
+                spanEndOffset -= 1
+                lastChar = runReadAction { currentRange.document.getText(TextRange(spanEndOffset - 1, spanEndOffset)) }
```

Comment:
Think this is cleaner:
```
            val lastIndexOfNewline = tempDocument.charsSequence.subSequence(currentRange.startOffset, currentRange.endOffset).lastIndexOf('\n')
            val spanEndOffset = if (lastIndexOfNewline != -1) lastIndexOfNewline else currentRange.endOffset
```

Document of `getText()` says for performance reason use `getCharsSequence()` whenever it's possible.

Reply:
so in this case, we assume there is at most 1 surfixing "\n"?

Reply:
but yes we can resolve this problem with the existing example without while loop

Reply:
ah then in case there are multiple you can do `.last { char -> char != '\n' }`